### ISSUE TITLE
PSResourceGet module assumes the required modules statement location in script file in specific place

### ISSUE THREAD

**Issue by peetrike** [PowerShell/PSResourceGet#1248]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

[About Requires](https://learn.microsoft.com/powershell/module/microsoft.powershell.core/about/about_requires) states that the #Requires statements can appear on any line in a script.  Yet the module PSResourceGet assumes that `#Requires -Modules` statement is above comment based help block.

```powershell
@'
<#PSScriptInfo
.VERSION 1.0.0
.GUID {0}
.AUTHOR Me
.TAGS
#>
<#
.DESCRIPTION
just for testing
#>

#Requires -Modules mymodule
'@ -f (new-guid) | set-content -path test.ps1

Get-PSScriptFileInfo -Path .\test.ps1 | select -ExpandProperty scriptRequiresComment
```

### Expected behavior

```console
RequiredModules
---------------
{mymodule}
```

### Actual behavior

```console
RequiredModules
---------------
{}
```

### Environment data

PSResourceGet: 0.5.22
PowerShell: 5.1 or 7.3.5

**Comment by peetrike** [comment]
`Publish-PSResource` makes the same assumption

**Comment by alerickson** [comment]
@anamnavi this should probably be tacked on with the others, let me know if you want me to look at this

**Comment by anamnavi** [comment]
@peetrike You're right that this differs from the docs. We can update this to mention how PSResourceGet differs. We would recommend that users place the `#Requires` comments before the help comment block (the one that includes `.DESCRIPTION`

From our code perspective, this was intentional. We assume the last "script metadata" like info will be within the help block. Anything after the closing bracket of the help block is "the rest of the file" to us. And we just grab those lines and dump it in a bucket until we reach the end of the file, without actually making meaning of that portion's contents.

So per our code the Requires has to come before the help block. If you wanted to, you could put the requires anywhere before that, but not in the script metadata block (ie the one that is <#PSScriptInfo ... #>)

**Comment by peetrike** [comment]
As this is different from PowerShellGet module and [About Requires](https://learn.microsoft.com/powershell/module/microsoft.powershell.core/about/about_requires) document, this should definitely be documented.  As well as location of script metadata block.  I assume that this is also expected to be before help block.

It would be nice, if `Test-PSScriptFileInfo` could explain, what exactly is non-conformant with script.  For example in Verbose or Warning stream.  Or when adding separate command-line parameter.

**Comment by anamnavi** [comment]
@sdwheeler we need to include this in the docs, thanks!

**Comment by sdwheeler** [comment]
@peetrike Note that for `Get-Help` to work for the comment-based help, you need a blank line between the comment blocks. This doesn't affect the PSResourceGet issue but I wanted to call it out.

**Comment by peetrike** [comment]
actually, it did affect PowerShellGet, when there was no blank line between script metadata and help block.  Not sure about `Get-Help`.  I just added _Description_ as this was required to publish script.
